tests/scripts/pulp_rpm/test_rpm_sync_publish.sh

```diff
@@ -49,7 +49,12 @@ expect_succ pulp rpm distribution create --name "cli_test_rpm_distro" \
   --base-path "cli_test_rpm_distro" \
   --publication "$PUBLICATION_HREF"
 DISTRIBUTION_BASE_URL=$(echo "$OUTPUT" | jq -r .base_url)
-expect_succ curl "$curl_opt" --head --fail "${DISTRIBUTION_BASE_URL}config.repo"
+if pulp debug has-plugin --name "rpm" --specifier ">=3.23.0"
+then
+  expect_fail curl "$curl_opt" --head --fail "${DISTRIBUTION_BASE_URL}config.repo"
+else
+  expect_succ curl "$curl_opt" --head --fail "${DISTRIBUTION_BASE_URL}config.repo"
+fi
```

Comment:
I think, you need to start with this to allow changing the expectation in 2.23.0.dev :

```suggestion
if pulp debug has-plugin --name "rpm" --specifier "<3.23"
then
  expect_succ curl "$curl_opt" --head --fail "${DISTRIBUTION_BASE_URL}config.repo"
fi
```